fedot/core/data/data.py

```diff
@@ -178,8 +205,30 @@ def from_csv(cls,
 
         features, target = process_target_and_features(df, target_columns)
 
-        return InputData(idx=idx, features=features, target=target, task=task, data_type=data_type,
-                         features_names=features_names)
+        categorical_features = None
+        if categorical_idx is not None:
+            if isinstance(categorical_idx[0], str):
+                raise
+            else:
+                categorical_idx = [i for i, column in enumerate(features_names) if column in set(categorical_idx)]
+
+            if isinstance(categorical_idx, list):
+                categorical_idx = np.array(categorical_idx)
+
+            categorical_features = features[:, categorical_idx].to_numpy()
+
+        data = InputData(
+            idx=idx,
+            features=features,
+            target=target,
+            task=task,
+            data_type=data_type,
+            features_names=features_names,
+            categorical_idx=categorical_idx,
+            categorical_features=categorical_features
+        )
```

Comment:
```suggestion
            if isinstance(categorical_idx[0], str):
                raise
            categorical_idx = [i for i, column in enumerate(features_names) if column in set(categorical_idx)]
            if len(categorical_idx) > 0:
                categorical_idx = np.array(categorical_idx)
                categorical_features = features[:, categorical_idx].to_numpy()
            else:
                categorical_idx = None

        data = InputData(
            idx=idx,
            features=features,
            target=target,
            task=task,
            data_type=data_type,
            features_names=features_names,
            categorical_idx=categorical_idx,
            categorical_features=categorical_features
        )
```

nit: кажется, ненужная проверка на список для `categorical_idx`. лучше проверить на всякий случай то, что вообще эти списки создаются (в противном случае передать в `InputData` значение `None`)

Reply:
можно, наверное, эту логику выделения np.array-ев с проверкой на то, что `categorical_idx` и `categorical_features` адекватно сформировались, вынести в отдельную приватную функцию

Reply:
понял, но в текущем варианте `categorical_idx` в любом случае будет списком, если я верно понимаю.

вот не вошли в первое ветвление с проверкой на `isinstance(categorical_idx[0], str)` не прокинули ошибку, потом ушли в генератор и получили список в любом случае

Reply:
@Lopa10ko упустил это место во время работы над ним. Я там собирался реализовать то, что если пользователь передает названия колонок, а не индексы и при этом название признаков забыл внести, то как раз вылезает ошибка. Если он указал, то мы преобразуем categorical_idx в индексы и в categorical_features кладем категориальные данные.

Иногда просто легче написать признаки названием, чем высчитывать их индексы

Reply:
@Lopa10ko изменения внес в [коммите](https://github.com/aimclub/FEDOT/pull/1318/commits/19e732e0dce6a8528c74e4faf69d296066ea75d0)